packages/gatsby/src/utils/jobs/manager.ts

```diff
@@ -132,15 +93,19 @@ async function runLocalWorker<T>(
   })
 }
 
+function isJobsIPCMessage(msg: any): msg is IncomingMessages {
+  return (
+    msg &&
+    msg.type &&
+    msg.payload &&
+    msg.payload.id &&
+    externalJobsMap.has(msg.payload.id)
+  )
+}
+
 function listenForJobMessages(): void {
   process.on(`message`, msg => {
-    if (
-      msg &&
-      msg.type &&
-      msg.payload &&
-      msg.payload.id &&
-      externalJobsMap.has(msg.payload.id)
-    ) {
+    if (isJobsIPCMessage(msg)) {
```

Comment:
jobs ipc messages are now typed now (this is code for external job runner), no change in behaviour